### PR TITLE
Fix ShellCheck warnings in /etc/bashrc

### PR DESCRIPTION
**Pull request by andersk**
###### Motivation for this change

https://github.com/koalaman/shellcheck/wiki/SC2086
Double quote to prevent globbing and word splitting.

https://github.com/koalaman/shellcheck/wiki/SC2166
Prefer `[ p ] && [ q ]` as `[ p -a q ]` is not well defined.

https://github.com/koalaman/shellcheck/wiki/SC2219
Instead of `let expr`, prefer `(( expr ))` .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/bash/bash.nix [hunk 1]
```diff
@@ -78,10 +78,10 @@ in
       promptInit = mkOption {
         default = ''
           # Provide a nice prompt if the terminal supports it.
-          if [ "$TERM" != "dumb" -o -n "$INSIDE_EMACS" ]; then
+          if [ "$TERM" != "dumb" ] || [ -n "$INSIDE_EMACS" ]; then
```
Comment (happysalada) [inline nixos/modules/programs/bash/bash.nix:81]: Do you think it's too different to switch from `[ .. -o .. ]` to `[[ .. || .. ]]` ?

Comment (andersk) [reply nixos/modules/programs/bash/bash.nix:81]: While `[ … -o … ]` is fundamentally unreliable due to a [grammatical ambiguity](https://rg1-teaching.mpi-inf.mpg.de/unixffb-ss98/quoting-guide.html#ssect:appl-test), I don’t have a preference between `[ … ] || [ … ]` and `[[ … || … ]]` except that the latter is non-portable (which isn’t a concern here), so I went with the one that’s more common.

Comment (happysalada) [reply nixos/modules/programs/bash/bash.nix:81]: Makes sense.

**Comment by happysalada** [comment]
Change looks good to me.
Leaving this one more day open in case anyone wants to have a look at it.

**Comment by happysalada** [comment]
Thank you for the contribution.
I forgot to say, but feel free to tag me if you need any review with shell stuff.

**Pull request merged**